Remove uses of CTFontCopyVariationAxesInternal in favor of the official API

vitorroriz opened the pull request:
#### d57febaa8b2ff0ea3570d7330f6728b859df5b7b
<pre>
Remove uses of CTFontCopyVariationAxesInternal in favor of the official API
<a href="https://bugs.webkit.org/show_bug.cgi?id=275977">https://bugs.webkit.org/show_bug.cgi?id=275977</a>
<a href="https://rdar.apple.com/130680716">rdar://130680716</a>

Reviewed by Brent Fulgham.

Since macOS 13.0 / iOS 16.00, we can get the variation axes with non localized axes names
by reading the kCTFontVariationAxesAttribute from font&apos;s descriptor.

We can now deprecate the SPI function.

* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::variationAxesWithNonLocalizedAxesNames):
(WebCore::variationAxes):
(WebCore::variationCapabilitiesForFontDescriptor):

Canonical link: <a href="https://commits.webkit.org/280537@main">https://commits.webkit.org/280537@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/333ebcfcf2818973003121c353c92d1827b49f54

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56731 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/36059 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9205 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60346 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/7174 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/58858 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43682 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7364 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45966 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/5039 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58761 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33897 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48968 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26823 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30677 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/6299 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6177 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/52621 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6570 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/62029 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/644 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6670 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/53224 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/647 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/49023 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/53243 "Found 1 new API test failure: /WebKitGTK/TestWebKitWebContext:/webkit/WebKitWebContext/memory-pressure (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12581 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/554 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31888 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32973 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/34058 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32719 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
```diff
@@ -57,24 +57,18 @@ bool fontNameIsSystemFont(CFStringRef fontName)
-    UNUSED_PARAM(shouldLocalizeAxisNames);
-    return adoptCF(CTFontCopyVariationAxes(font));
-#endif
+    // Reading kCTFontVariationAxesAttribute returns non localized axes names
```
Great comment!

brentfulgham commented on Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
```diff
@@ -57,24 +57,18 @@ bool fontNameIsSystemFont(CFStringRef fontName)
 }
 
-static RetainPtr<CFArrayRef> variationAxes(CTFontRef font, ShouldLocalizeAxisNames shouldLocalizeAxisNames)
+static RetainPtr<CFArrayRef> variationAxesWithNonLocalizedAxesNames(CTFontDescriptorRef fontDescriptor)
```
Nit: The changes here have the effect of swapping the order of these functions. I understand that this first function is now only needed when `USE(KCTFONTVARIATIONAXESATTRIBUTE)` is true (so you moved it into the macro below), but it makes the diff a little harder to follow (to see before/after).

vitorroriz replied:
oh yeah, it does make it harder to read, sorry. I've actually moved it because `variationAxesWithNonLocalizedAxesNames` is now used by `variationAxes`, so I wanted to avoid having to declare the static function before.

brentfulgham replied:
Ah, that makes sense. No need to change the ordering.

brentfulgham reviewed (approved):
I would change the title of the PR and bug:

"Remove uses of CTFontCopyVariationAxesInternal in favor of the official API"

Constellation reviewed (changes requested):
Can you add a link to A/B test results on radar etc.?
Also, can you update the commit message?

vitorroriz committed:
Remove uses of CTFontCopyVariationAxesInternal in favor of the official API
https://bugs.webkit.org/show_bug.cgi?id=275977
rdar://130680716

Reviewed by Brent Fulgham.

Since macOS 13.0 / iOS 16.00, we can get the variation axes with non localized axes names
by reading the kCTFontVariationAxesAttribute from font's descriptor.

We can now deprecate the SPI function.

* Source/WebCore/platform/graphics/cocoa/FontCacheCoreText.cpp:
(WebCore::variationAxesWithNonLocalizedAxesNames):
(WebCore::variationAxes):
(WebCore::variationCapabilitiesForFontDescriptor):

Canonical link: https://commits.webkit.org/280537@main
